tests/nested/manual/core20-set-efi-boot-vars/task.yaml

```diff
@@ -0,0 +1,177 @@
+    BootOrder variable should be the corresponding number.
+
+systems: [ubuntu-20.04-64, ubuntu-22.04-64]
+
+environment:
+    NESTED_ENABLE_TPM: true
+    NESTED_ENABLE_SECURE_BOOT: true
+    NESTED_BUILD_SNAPD_FROM_CURRENT: true
+
+    DESCRIPTION/A: "Install with assets in EFI/boot/, then reinstall to assets in EFI/ubuntu/."
+    DESCRIPTION/B: "Install with assets in EFI/ubuntu/, then reinstall to assets in EFI/boot/."
+
+    INITIAL_ASSET_DIR/A: "boot"
+    INITIAL_ASSET_DIR/B: "ubuntu"
+
+    FINAL_ASSET_DIR/A: "ubuntu"
+    FINAL_ASSET_DIR/B: "boot"
+
+prepare: |
+    snap install yq
+
+    VERSION=$(tests.nested show version)
+    echo "Download pc-gadget to use in initial image"
+    snap download --basename=pc --channel="$VERSION/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    ARCH=$(find pc-gadget -name 'grub*.efi' -printf '%f\n' | sed 's/grub//;s/\.efi//')
+
+    case "${ARCH}" in
+        x64  ) ;;
+        aa64 ) ;;
+        *    ) ERROR "Invalid architecture '${ARCH}': must be 'x64' or 'aa64'" ;;
+    esac
+
+    echo "Set up gadget with EFI assets in EFI/${INITIAL_ASSET_DIR}"
+    case "${INITIAL_ASSET_DIR}" in
+        boot )
+            yq -i '(.volumes.pc.structure[] | select(.role == "system-seed") | .content) |= [{"source": "grub'"${ARCH}"'.efi", "target": "EFI/boot/grub'"${ARCH}"'.efi"}, {"source": "shim.efi.signed", "target": "EFI/boot/boot'"${ARCH}"'.efi"}]' pc-gadget/meta/gadget.yaml
+            ;;
+        ubuntu )
+            yq -i '(.volumes.pc.structure[] | select(.role == "system-seed") | .content) |= [{"source": "grub'"${ARCH}"'.efi", "target": "EFI/ubuntu/grub'"${ARCH}"'.efi"}, {"source": "shim.efi.signed", "target": "EFI/ubuntu/shim'"${ARCH}"'.efi"}, {"source": "fb'"${ARCH}"'.efi", "target": "EFI/boot/fb'"${ARCH}"'.efi"}]' pc-gadget/meta/gadget.yaml
+            ;;
+        * )
+            ERROR "Invalid initial asset dir: ${INITIAL_ASSET_DIR}"
+            ;;
+    esac
+
+    echo "Make sure fb${ARCH}.efi is empty so it can't set EFI variables"
+    truncate -s 0 "pc-gadget/fb${ARCH}.efi"
+
+    echo "Get snakeoil key"
+    KEY_NAME=$(tests.nested download snakeoil-key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+    echo "Sign the modified pc-gadget"
+    tests.nested secboot-sign gadget pc-gadget "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+
+    echo "Repack the modified pc-gadget"
+    snap pack pc-gadget/ "$(tests.nested get extra-snaps-path)"
+    # No need to re-sign again after repacking
+    echo "Build core image around modified pc-gadget"
+    tests.nested build-image core
+    echo "Create VM around custom image"
+    tests.nested create-vm core
+
+execute: |
+    ARCH=$(find pc-gadget -name 'grub*.efi' -printf '%f\n' | sed 's/grub//;s/\.efi//')
+
+    case "${ARCH}" in
+        x64  ) ;;
+        aa64 ) ;;
+        *    ) ERROR "Invalid architecture '${ARCH}': must be 'x64' or 'aa64'" ;;
```

Comment:
Why do we test for that?

Reply:
I just want to make sure the `ARCH` variable was set correctly, otherwise later parts of the test fail in not fun to debug ways. But it's true, maybe a `test -n $ARCH` would be better.

Reply:
Alternatively, if there's a better way to compute the ARCH string used in EFI assets (i.e. the `x64` in `shimx64.efi`), that would be even better. In addition to being set, I also wanted to make sure that the logic which sets the `ARCH` variable was successful.